lib/phoenix_live_view/diff.ex

```diff
@@ -356,7 +356,7 @@ defmodule Phoenix.LiveView.Diff do
     %{
       socket
       | assigns: assigns,
-        private: private,
+        private: private |> Enum.into(Map.take(socket.private, [:conn_session])),
```

Comment:
We only need to do this on line #349. You can make the third argument be `Map.take(socket.private, [:conn_session])`. Can you please give it a try?